app/src/main/java/com/fish/monsters/features/preferences/PreferencesManager.kt

```diff
@@ -0,0 +1,26 @@
+package com.fish.monsters.features.safetyInformation
+
+import android.content.Context
+import androidx.datastore.preferences.preferencesDataStore
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import kotlinx.coroutines.flow.map
+
+val Context.dataStore by preferencesDataStore(name = "settings")
```

Comment:
Data store będzie działał w całej apce, przenieś zatem tą linie do osbnego pliku może to być plik w `/core/database` lub `/common/extensions` jako `ContextExt.kt`. Następnie dodaj data store do DI.

```
val appModule = module {
    // [...] reszta inicjalizacji
    single { androidContext().dataStore }
}
```